google/cloud/pubsub/internal/message_carrier.h

```diff
@@ -0,0 +1,63 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifdef GOOGLE_CLOUD_CPP_HAVE_OPENTELEMETRY
```

Comment:
These conditionals should probably be within the include guards.

Reply:
It doesn't make a difference either way, imo.

This file shouldn't exist if it isn't compiled with out otel. 

I don't care too much though. So changed.

Reply:
> It doesn't make a difference either way, imo.

Not in this case, but that doesn't mean it is best practice, nor that it matches what we do elsewhere.  Consider what would need to happen should such a file grow a `#else  // GOOGLE_CLOUD_CPP_HAVE_OPENTELEMETRY` in the future, for example.  It also defeats any preprocessor smarts to make include guards as efficient as `#pragma once`.